src/modules/configurator/hooks/useCarImageSlide.tsx

```diff
@@ -0,0 +1,42 @@
+import { useRef, useState, useEffect } from "react";
+import { fetchCarImagesByColorAndVariant } from "../../../services/API_configurations";
+import { userConfigurationState } from "../state";
+import { useRecoilValue } from "recoil";
+
+export function useCarImageSlide() {
+  const [configImages, setConfigImages] = useState<string[]>();
+  const currentUserConfiguration = useRecoilValue(userConfigurationState);
+
+  const paginationBackRef = useRef(null);
+  const paginationNextRef = useRef(null);
+
+  useEffect(() => {
+    if (currentUserConfiguration) {
+      handleCarImageFetch(
+        currentUserConfiguration.model,
+        currentUserConfiguration.color.label,
+        currentUserConfiguration.wheels.label
+      );
+    }
+  }, [currentUserConfiguration]);
+
+  const handleCarImageFetch = async (
+    modelName: "Audi RS5" | "Audi RS6" | "Audi e-tron GT",
```

Comment:
If you are using a type on more than 1 place, than create a global type.

Reply:
Extracted to a global type.